plugins/modules/activation_key.py

```diff
@@ -47,21 +47,28 @@
     type: str
   subscriptions:
     description:
-      - List of subscriptions that include either Name or Pool ID.
-      - Pool IDs are preferred since Names are not unique and the module will fail if it finds more than one subscription with the same name.
+      - List of subscriptions that include either Name, Pool ID, or Upstream Pool ID.
+      - Pool IDs are preferred since Names and Upstream Pool IDs are not guaranteed to be unique. The module will fail if it finds more than one match.
     type: list
     elements: dict
     suboptions:
       name:
         description:
           - Name of the Subscription to be added.
-          - Mutually exclusive with I(pool_id).
+          - Mutually exclusive with I(pool_id) and I(upstream_pool_id).
         type: str
         required: false
       pool_id:
         description:
           - Pool ID of the Subscription to be added.
-          - Mutually exclusive with I(name).
+          - Mutually exclusive with I(name) and I(upstream_pool_id).
+        type: str
+        required: false
+      upstream_pool_id:
+        description:
+          - Upstream Pool ID of the Subscription to be added.
+          - Mutually exclusive with I(name) and I(pool_id).
+          - Supported since Katello 3.15.0
```

Comment:
I must admit I prefer not to have version numbers in doc strings, for two reasons:
* upstream moves quickly (heck, 3.15 is not supported anymore ;)) and this makes the comment kinda out of date quickly
* downstream the users don't necessarily know which component versions they have.

I wonder if there is a trivial way to deduct support for that param from the apidoc and warn the user in the case it's not supported.

either way, shouldn't block this pr